### PR TITLE
add manifests to deploy toolpad on kubernetes

### PR DESCRIPTION
**Pull request by chanwit**
Signed-off-by: Chanwit Kaewkasi <chanwit@gmail.com>

Closes https://github.com/mui/mui-toolpad/issues/1495

Deployment tested on a local KIND cluster.
image

How to test this PR:
- `kind create cluster`
- `kubectl apply -f kubernetes/toolpad.yaml`
- `kubectl -n toolpad port-forward deployment/toolpad 3000:3000`
- Browse http://localhost:3000

### CODE REVIEW

**Comment by Janpot** [comment]
Hi @chanwit, thanks for your contribution. In order for us to merge this, we'd need a few more changes:

1. Currently our CI is failing over some formatting issues in the yaml file. You can fix those by running `yarn prettier` in the root of the project
2. Ideally we'd add a section "With Kubernetes" under https://mui.com/toolpad/getting-started/installation/. The prerequisites can be: access to a cluster, and the kubernetes CLI installed. the steps can be: download config files, run `kubectl apply ...`

**Comment by chanwit** [comment]
Thank you @Janpot 
I'm working on it (also on the docs)

**Comment by chanwit** [comment]
kindly ping @Janpot

File: docs/data/toolpad/getting-started/installation.md [hunk 1]
```diff
@@ -25,3 +25,39 @@
    ```
 
 1. Build and deploy applications on http://localhost:3000/.
+
+## With Kubernetes
+
+### Prerequisites
+
+- A Kubernetes cluster (a local one via `kind` or a remote one)
+- The `kind` command line tool installed on your machine. Follow the [official instructions](https://kind.sigs.k8s.io/docs/user/quick-start/) to install `kind`.
```
Comment (Janpot) [inline docs/data/toolpad/getting-started/installation.md:34]: This should be agnostic from `kind`. "a cluster" is a prerequisite. We can give suggestions on how to create/obtain a cluster, but we shouldn't be opinionated about this

File: docs/data/toolpad/getting-started/installation.md [hunk 2]
```diff
@@ -25,3 +25,39 @@
...
+
+### Steps
+
+1. Download the Toolpad Kubernetes manifest file:
+
+   ```sh
+   curl -LO https://raw.githubusercontent.com/mui/mui-toolpad/master/kubernetes/toolpad.yml
+   ```
+
+1. Create a Kubernetes cluster with `kind`:
```
Comment (Janpot) [inline docs/data/toolpad/getting-started/installation.md:45]: Having access to a cluster is already a prerequisite

File: kubernetes/toolpad.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
+---
```
Comment (Janpot) [inline kubernetes/toolpad.yaml:1]: Can this be placed under `/deploy/kubernetes/toolpad.yaml`?

**Comment by Janpot** [comment]
Closing as per https://github.com/mui/mui-toolpad/discussions/1748
Nonetheless, thank you for the effort, I'm sorry in the end this didn't make it in.

**Pull request closed without merging**